packages/api/controllers/user.js

```diff
@@ -20,4 +20,34 @@ const getUserById = async (req, res) => {
   }
 };
 
-module.exports = { getUserById };
+const create = async (req, res) => {
+  const { email, username } = req.body;
+  if (!email) {
+    return res.status(400).send('Missing email');
+  }
+  if (!username) {
+    return res.status(400).send('Missing username');
+  }
+
+  try {
+    const [user, created] = await db.User.findOrCreate({
+      where: {
+        email,
+      },
+      defaults: {
+        username, email,
+      },
+    });
+    if (user) {
+      if (created) return res.status(201).json({ user });
+      return res.status(400).send('User already exists!');
+    }
+    console.info('unknown 400 error @ POST /user');
+    return res.status(400).send('Something went wrong');
```

Comment:
A `4XX` error indicates the user did something wrong that they can correct. For example, a 400 suggests they can change the body of the request. A 401 suggests they can use authentication to complete the request. A 409 (in this case) might suggest they use a different email address.

What (if anything) could the user change if they reach this point, and how can we message that?